### PR TITLE
Add support for shadow modifier

### LINKED ISSUE #324: Shadow support
```swift
public func shadow(
  color: Color = Color(.sRGBLinear, white: 0, opacity: 0.33),
  radius: CGFloat,
  x: CGFloat = 0, y: CGFloat = 0
) -> some View
```

> Adds a shadow to this view.
>
> The example below a series shows of boxes with increasing degrees of shadow ranging from 0 (no shadow) to 5 points of shadow, offset down and to the right of the views:
> 
> ```swift
> struct Shadow: View {
>     var body: some View {
>         HStack {
>             ForEach(0..<6) {
>                 Color.red.frame(width: 60, height: 60, alignment: .center)
>                     .overlay(Text("\($0)"),
>                              alignment: .bottom)
>                     .shadow(color: Color.gray,
>                             radius: 1.0,
>                             x: CGFloat($0),
>                             y: CGFloat($0))
>             }
>         }
>     }
> }
> ```
> 
> - Parameters:
>   - color: The shadow's color.
>   - radius: The shadow's size.
>   - x: A horizontal offset you use to position the shadow relative to
>     this view.
>   - y: A vertical offset you use to position the shadow relative to this
>     view.
> - Returns: A view that adds a shadow to this view.

### PR DESCRIPTION
**Pull request by literalpie**
Closes #324

### CODE REVIEW

File: Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift [hunk 1]
```diff
@@ -101,3 +101,11 @@ extension _PaddingLayout: DOMViewModifier {
       .joined(separator: " ")]
   }
 }
+
+extension _ShadowLayout: DOMViewModifier {
+  public var attributes: [HTMLAttribute: String] {
+    ["style": "box-shadow: \(x)px \(y)px \(radius)px 0px \(color.cssValue(.defaultEnvironment));"]
```
Comment (carson-katri) [inline Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift:107]: Maybe you can multiply the radius by 2? Seems like when going from [Sketch to UIKit](https://sarunw.com/posts/match-view-shadow-to-sketch-shadow/#blur-vs.-shadow-radius) (which uses blur and spread), they'll divide by two, so the inverse could maybe work here?

Comment (literalpie) [reply Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift:107]: Yes, that's at least closer than what I had (color meter says it's not a match, but it's close enough for my naked eye)

File: Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift [hunk 2]
```diff
@@ -101,3 +101,13 @@ extension _PaddingLayout: DOMViewModifier {
       .joined(separator: " ")]
   }
 }
+
+extension _ShadowLayout: DOMViewModifier {
+  public var attributes: [HTMLAttribute: String] {
+    [
+      "style": "box-shadow: \(x)px \(y)px \(radius * 2)px 0px \(color.cssValue(.defaultEnvironment));",
```
Comment (literalpie) [inline Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift:108]: I almost forgot about this `defaultEnvironment`. I'm not sure how to make this come from the app's actual environment.

Comment (carson-katri) [reply Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift:108]: I think you can add `@Environment(\.self) var environmentValues` to the modifier to access it.

**Comment by MaxDesiatov** [comment]
@literalpie to fix the macOS build you'll have to open the demo Xcode project (in the `NativeDemo` subdirectory) and add the new `ShadowDemo` file to it.

File: Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift (hunk 2 above)
Comment (literalpie) [reply Sources/TokamakStaticHTML/Modifiers/LayoutModifiers.swift]: I wasn't able to use `@Environment(\.self)`. I used `EnvironmentReader` which is how this is done in other modifiers (specifically `_BackgroundModifier`).

**Review by MaxDesiatov — APPROVED** [review]
Awesome, LGTM 👍

**Review by carson-katri — APPROVED** [review]

**Pull request merged**